### PR TITLE
add argument checkpoint-freq to classification reference scripts

### PR DESCRIPTION
**Pull request by JonnaMat**
Adds a new argument `--output-freq` to the parser.
Adds functionality of specifying the output/checkpoint frequency during training.

Closes https://github.com/pytorch/vision/issues/7481

### CODE REVIEW

File: references/classification/train.py [hunk 1]
```diff
@@ -359,7 +359,7 @@ def collate_fn(batch):
         evaluate(model, criterion, data_loader_test, device=device)
         if model_ema:
             evaluate(model_ema, criterion, data_loader_test, device=device, log_suffix="EMA")
-        if args.output_dir:
+        if args.output_dir and (epoch + 1) % args.output_freq == 0:
```
Comment (NicolasHug) [inline references/classification/train.py:362]: To be make sure we always save the last epoch as well, we might want to add:

```suggestion
        if args.output_dir and ((epoch + 1) % args.output_freq == 0) or epoch == args.epochs - 1):
```

File: references/classification/train.py [hunk 2]
```diff
@@ -440,6 +440,7 @@ def get_args_parser(add_help=True):
     parser.add_argument("--lr-min", default=0.0, type=float, help="minimum lr of lr schedule (default: 0.0)")
     parser.add_argument("--print-freq", default=10, type=int, help="print frequency")
     parser.add_argument("--output-dir", default=".", type=str, help="path to save outputs")
+    parser.add_argument("--output-freq", default=1, type=int, help="output frequency")
```
Comment (NicolasHug) [inline references/classification/train.py:443]:
```suggestion
    parser.add_argument("--checkpoint-freq", default=1, type=int, help="Create a checkpoint every checkpoint-freq epochs")
```

Comment (JonnaMat) [reply references/classification/train.py:443]: I agree, that `--checkpoint-freq` makes more sense but I wanted to stay consistent with `--output-dir`, perhaps changing `output-dir` to `checkpoint-dir`?

Comment (JonnaMat) [reply references/classification/train.py]: All right! Changed it now to checkpoint-freq as suggested.

**Review by NicolasHug — APPROVED** [review]
Thank you for the PR @JonnaMat !

**Comment by NicolasHug** [comment]
@JonnaMat I'm trying to merge your PR but for that we need it to be in sync with the current `main` branch. It looks like you have restricted the rights for me to push on your banch so I cannot update the PR manually. Would you mind follow the steps in https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork to allow me to push and update? (this would only give me write permission to this specific branch, not to your repo).

The alternative is for you to manually sync with `main` but that may be harder because we constantly update `main` and this PR may become outdated quite fast.

**Comment by JonnaMat** [comment]
@NicolasHug The document you linked only mentioned to check "Allow edits by maintainers" which is done. 
Let me know if you need any other permissions. :)

**Comment by NicolasHug** [comment]
Something seems to be off. I can see the " Maintainers are allowed to edit this pull request. " text on the right. However I still can't push to your branch and I also can't update it from the GitHub GUI. Using `gh pr` and pusing I'm getting:


```
(pt) ➜  vision git:(feature/clf-train-script-checkpointing) ✗ git push 
ERROR: Permission to JonnaMat/vision.git denied to NicolasHug.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I asked @pmeier  to check and he's experience similar permission errors. This is the first time I'm seeing this on a PR. @JonnaMat  did you setup your repo in a specific way that restricts permissions in some way?

**Pull request closed without merging**